tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -13530,6 +13527,13 @@ void SpirvEmitter::processSwitchStmtUsingSpirvOpSwitch(
     doDeclStmt(condVarDeclStmt);
 
   auto *cond = switchStmt->getCond();
+  if (llvm::dyn_cast<IntegerLiteral>(cond)) {
+    emitError(
+        "integer literal selectors in switch statements not yet implemented",
+        cond->getLocStart());
+    return;
+  }
```

Comment:
Under which conditions does this error come up? I see an error as a condition that could happen if the programmer did something wrong. There should be a test that these are emitted correctly. An assert or llvm unreachable is a case that should never happen. If it does, then there is something wrong in the code.

Please add a test for this error.

Reply:
Test added. It used to fail with `error: non-32bit integer case value in switch statement unimplemented` which was a bit confusing since it was a 32-bit integer literal. I haven't gotten to actually implementing that, but that was my original motivation that led me to this fix first on the way there.